src/js.cookie.mjs

```diff
@@ -9,11 +9,27 @@ function extend () {
   return result
 }
 
-function decode (s) {
-  return s.replace(/(%[\dA-F]{2})+/gi, decodeURIComponent)
+var rfc6265Converter = {
+  read: function (value, key) {
+    return value.replace(/(%[\dA-F]{2})+/gi, decodeURIComponent)
+  },
+  write: function (value, key) {
+    return encodeURIComponent(value).replace(
+      /%(23|24|26|2B|3A|3C|3E|3D|2F|3F|40|5B|5D|5E|60|7B|7D|7C)/g,
+      decodeURIComponent
+    )
+  }
 }
 
 function init (converter) {
+  // normalize converter
+  if (!converter.read) {
+    converter = {
+      read: converter,
+      write: converter.write || rfc6265Converter.write
+    }
+  }
```

Comment:
Another note: for v3 we may want to change the `withConverter()` api slightly. I don't see a reason why `withConverter(func)` should turn the given func into the read converter. It could just as well have been the write converter. I'd prefer to make passing the type of converter explicit, seems simpler, not only for us to implement, but in the end also for a user to remember.

This would then allow to simplify this to (I believe):

```javascript
function init (converter) {
  converter = extend(rfc6265Converter, converter);
  ...
```
```

```javascript
converter = ({read: converter, write
```

Reply:
I agree, this would make the API clearer and would simplify the internal logic.

Reply:
=> #570

Reply:
👍 to remove `withConverter(fn)`. We did that at the time because we didn't have a use case to write. Then we have so it becomes pointless.